ui/src/app/components/text-modal.tsx

```diff
@@ -20,6 +20,7 @@ export interface TextModalProps {
   body: string;
   buttonText: string;
   closeFunction: Function;
+  role?: 'dialog' | 'alertdialog';
```

Comment:
Is this an ADT?  very cool.  How does this compare to using an Enum?

Reply:
Not sure, please ask offline so we can discuss.